channeldb/db.go

```diff
@@ -216,7 +216,7 @@ func fileExists(path string) bool {
 // associated with the target nodeID. In the case that no active channels are
 // known to have been created with this node, then a zero-length slice is
 // returned.
-func (d *DB) FetchOpenChannels(nodeID *btcec.PublicKey) ([]*OpenChannel, error) {
+func (d *DB) FetchOpenChannels(nodeID *btcec.PublicKey, includePendingOpen, includeOpen, includePendingClose, includeClosed bool) ([]*OpenChannel, error) {
```

Comment:
I'd say a channel that has been closed, shouldn't be possibly returned by a method called `FetchOpenChannels`. Instead, a new internal function that takes a predicate argument can be used to implement a general or specific method that returns a set of channels in the desired state.